core/src/main/java/org/prevayler/Prevayler.java

```diff
@@ -33,7 +33,7 @@
    *
    * @see org.prevayler.PrevaylerFactory
    */
-  public void execute(Transaction<? super P> transaction);
```

Comment:
Do you have a specific reason for removing `? super` throughout this interface? It will cause problems for some users.

For a contrived example, imagine you have a `Prevayler<ArrayList<String>>` and want to execute a `Transaction<List<String>>` on it. That should logically be possible, and the `? super` allowed it to be done.

Reply:
Yes, I think you are right. My bad... I haven't thought this through completely. I was misleaded by the fact that eg in Java 8 there is a trimSize() method in ArrayList which doesn't exist in List. Therefore I was assuming that we could potentially run into problems when a List was passed in as parameter where an ArrayList was expected, which would cause problems if the corresponding code would actually try to call trimToSize() on it. But - as a Transaction is self-contained and type-consistent such a case cannot be constructed without the use of explicit unsafe casts. Any of List's methods the "limited" transaction can possibly call will also exist in ArrayList per definition.
Sorry if I am being overly verbose here, but maybe it will be more comprehensive to someone else reading this and trying to follow our thoughts. I generally feel that real-world use-cases for "? super X" constructs are harder to find and a lot less obvious than "? extends X" ones.

Reply:
Done.